spark/src/test/scala/org/apache/spark/sql/delta/deletionvectors/DeletionVectorsSuite.scala

```diff
@@ -791,4 +810,23 @@ object DeletionVectorsSuite {
   val table5Sum = 21975159654L
   val table5CountByValues = (0 to 20).map(_ -> 99900000L).toMap + (21 -> 47436174L)
   val table5SumByValues = (0 to 20).map(v => v -> v * 99900000L).toMap + (21 -> 21 * 47436174L)
+
+  // A template table with special characters in DV path. To use this table, one must use
+  // the `withTableHavingSpecialCharInDVPath` method to create a copy of this table with paths
+  // filled.
+  // Content of this table is range(0, 10) with all even numbers deleted.
+  val tableHavingSpecialCharInDVTemplate = "src/test/resources/delta/table-with-dv-special-char"
+
+  def writeTableHavingSpecialCharInDVPath(path: File, pathIsEncoded: Boolean): Unit = {
+    FileUtils.copyDirectory(new File(tableHavingSpecialCharInDVTemplate), path)
+    val fullPath = new File(
+      path,
+      if (pathIsEncoded) "folder%26with%25special%20char" else "folder&with%special char")
+      .getCanonicalPath
+    val logJson = new File(path, "_delta_log/00000000000000000000.json")
+    val logJsonContent = FileUtils.readFileToString(logJson, "UTF-8")
+    val newLogJsonContent = logJsonContent.replace(
+      "{{FOLDER_WITH_SPECIAL_CHAR}}", fullPath)
+    FileUtils.write(logJson, newLogJsonContent, "UTF-8")
```

Comment:
Since we are manually manipulating the JSON, can we read back the final JSON and validate that it correctly formed?